src/HotChocolate/AspNetCore/src/AspNetCore/Subscriptions/Subscription.cs

```diff
@@ -39,51 +42,65 @@ internal sealed class Subscription : ISubscription
-                        await _connection.SendAsync(new DataResultMessage(Id, result), _cts.Token);
+                        if (!cancellationToken.IsCancellationRequested && !_connection.Closed)
+                        {
+                            await _connection.SendAsync(
+                                new DataResultMessage(Id, result),
+                                cancellationToken);
+                        }
                     }
                 }
 
-                if (!_cts.IsCancellationRequested)
+                if (!cancellationToken.IsCancellationRequested && !_connection.Closed)
                 {
-                    await _connection.SendAsync(new DataCompleteMessage(Id), _cts.Token);
-                    Completed?.Invoke(this, EventArgs.Empty);
+                    await _connection.SendAsync(new DataCompleteMessage(Id), cancellationToken);
                 }
             }
-            catch (OperationCanceledException) { }
+            catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested) { }
             catch (ObjectDisposedException) { }
-            catch (Exception ex)
+            catch (Exception ex) when (!cancellationToken.IsCancellationRequested)
             {
-                if (!_cts.IsCancellationRequested)
+                if (!_connection.Closed)
                 {
-                    IError error =
-                        ErrorBuilder
-                            .New()
-                            .SetException(ex)
-                            .SetCode(ErrorCodes.Execution.TaskProcessingError)
-                            .SetMessage("Unexpected Execution Error")
-                            .Build();
-
-                    IQueryResult result = QueryResultBuilder.CreateError(error);
                     try
                     {
-                        await _connection.SendAsync(new DataResultMessage(Id, result), _cts.Token);
+                        try
+                        {
+                            await _connection.SendAsync(
+                                new DataResultMessage(Id, UnknownSubscriptionError(ex)),
+                                cancellationToken);
+                        }
+                        finally
+                        {
+                            await _connection.SendAsync(
+                                new DataCompleteMessage(Id),
+                                cancellationToken);
+                        }
                     }
-                    finally
+                    catch
                     {
-                        await _connection.SendAsync(new DataCompleteMessage(Id), _cts.Token);
-                        Completed?.Invoke(this, EventArgs.Empty);
+                        // suppress all errors, so original exception can be rethrown
                     }
                 }
+
+                // original exception should be propagated to upper level in order to be logged
```

Comment:
We will remove this in favor of proper diagnostic events. I will add a new diagnostic event tomorrow and we will pass the through the the diagnostic events.